packages/shopify/src/utilities/memoize.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Creates a memoized version of an async function with a custom cache key generator.
+ * Only successful results are cached - rejected promises are not remembered.
+ *
+ * @param fn - The async function to memoize
+ * @param getCacheKey - Function that generates a cache key from the function arguments
+ * @returns Object with the memoized function and cache clearing capability
+ */
+export function memoize<TArgs extends unknown[], TReturn>(
```

Comment:
there is no version of this available from NPM?

Reply:
There is, and normally i would include that, however the conventions in this codebase show a strong preference for implementing your own helper functions rather than adding dependencies. I think this is to avoid bloat, risk etc. and because customers are running this code on their storefronts.